src/signal.rs

```diff
@@ -2305,3 +2305,224 @@ mod cell_renderer {
+        fn connect_changed<F>(&self, changed_func: F) -> u64
+        where F: Fn(&Self) + 'static {
+            unsafe {
+                let f: Box<Box<Fn(&Self) + 'static>> =
+                    Box::new(Box::new(changed_func));
+                connect(self.to_glib_none().0, "changed",
+                    transmute(trampoline::<Self> as usize), Box::into_raw(f) as *mut _)
+            }
+        }
+
+        fn connect_delete_text<F>(&self, delete_text_func: F) -> u64
+        where F: Fn(&Self, i32, i32) + 'static {
+            unsafe {
+                let f: Box<Box<Fn(&Self, i32, i32) + 'static>> =
+                    Box::new(Box::new(delete_text_func));
+                connect(self.to_glib_none().0, "delete-text",
+                    transmute(delete_trampoline::<Self> as usize), Box::into_raw(f) as *mut _)
+            }
+        }
+
+        fn connect_insert_text<F>(&self, insert_text_func: F) -> u64
+        where F: Fn(&Self, &str, &mut u32) + 'static {
+            unsafe {
+                let f: Box<Box<Fn(&Self, &str, &mut u32) + 'static>> =
+                    Box::new(Box::new(insert_text_func));
+                connect(self.to_glib_none().0, "insert-text",
+                    transmute(insert_trampoline::<Self> as usize), Box::into_raw(f) as *mut _)
+            }
+        }
+    }
+
+    unsafe extern "C" fn trampoline<T>(this: *mut GtkEditable,
+                                       f: &Box<Fn(&T) + 'static>)
+    where T: IsA<Editable> {
+        callback_guard!();
+        f(&Editable::from_glib_none(this).downcast_unchecked());
+    }
+
+    unsafe extern "C" fn delete_trampoline<T>(this: *mut GtkEditable,
+                                              start_pos: c_int,
+                                              end_pos: c_int,
+                                              f: &Box<Fn(&T, i32, i32) + 'static>)
+    where T: IsA<Editable> {
+        callback_guard!();
+        f(&Editable::from_glib_none(this).downcast_unchecked(), start_pos, end_pos);
+    }
+
+    unsafe extern "C" fn insert_trampoline<T>(this: *mut GtkEditable,
+                                              new_text: *mut c_char,
+                                              new_text_length: c_int,
+                                              position: *mut c_uint,
+                                              f: &Box<Fn(&T, &str, Option<&mut u32>) + 'static>)
+    where T: IsA<Editable> {
+        callback_guard!();
+        if new_text_length != -1 {
+            sub_func(this,
+                     Vec::from_raw_parts(new_text as *mut c_uchar,
+                                         new_text_length as usize,
+                                         new_text_length as usize).as_slice(),
+                     position,
+                     f);
+        } else {
+            sub_func(this, CStr::from_ptr(new_text).to_bytes(), position, f);
+        }
+    }
+
+    unsafe fn sub_func<T>(this: *mut GtkEditable,
```

Comment:
It was for a borrow issue. It'll be solved when I'll replace `Vec::from_raw_parts` by `slice::from_raw_parts`.